homeassistant/components/vitrea/__init__.py

```diff
@@ -77,8 +78,11 @@ def handle_new_entity(event) -> None:
         max_discovery_time = 90  # a typical 20 nodes setup can take up to 60 seconds
         discovery_start = time.monotonic()
 
+        # Use shorter intervals to satisfy testing
+        sleep_interval = 4 # could also do 0.1 if "pytest" in sys.modules else 3
+
         while True:
-            await asyncio.sleep(5)
+            await asyncio.sleep(sleep_interval)
```

Comment:
It's not OK to sleep for longer than a tenth of a second in tests.

Normally you shouldn't need to sleep longer than 0 seconds. It's enough to yield to the event loop.

Take a different approach in the test. Eg patch the timeout to 0, if we want to test a timeout.